server/views/pages/printNotification/printNotificationSlip.njk

```diff
@@ -0,0 +1,128 @@
+                     href="{{ printLinks }}&pageType=offender">
+                     Print notification slip
+                  </a>
+
+                  <a data-qa="slip-establishment-copy" data-prevent-double-click="true" class="govuk-button govuk-button--secondary no-js-hidden" data-module="govuk-button"
+                     href="{{ printLinks }}&pageType=establishment">
+                     Print establishment copy
+                  </a>
+              </div>
+            </div>
+          </div>
+
+          <div class="notification-slip">
+
+           {% if pageType == 'offender' %}
+              <div data-qa="page-title-caption" class="offender-title">[Offender copy]</div>
+            {% elif pageType == 'establishment' %}
+              <div data-qa="page-title-caption" class="establishment-title">[Establishment copy]</div>
+            {% endif %}
+
+            <div class="govuk-grid-row">
+              <div class="govuk-grid-column-full">
+                <span class="govuk-caption-m" data-qa="prison-name">
+                  {% if model.prisonerDetail.assignedLivingUnit.agencyName %}
+                    {{ model.prisonerDetail.assignedLivingUnit.agencyName }}
+                  {% else %}
+                    No agency name available
+                  {% endif %}
+                </span>
+                <h1 class="govuk-heading-l" data-qa="page-title">Release date notification slip</h1>
+              </div>
+            </div>
+
+            {{ prisonerDetails(model) }}
+
+            <div class="govuk-grid-row">
+              <div class="govuk-grid-column-full">
+                <p class="govuk-body" data-qa="calculation-date">These release dates were calculated on {{ calculationDate | date('DD MMMM YYYY') }}</p>
+
+                {% if pageType == 'establishment' %}
+                    <dl class="govuk-summary-list govuk-summary-list--no-border govuk-!-margin-bottom-4">
+                        <div class="govuk-summary-list__row">
+                            <dt data-qa="calculation-reason-title" class="govuk-summary-list__key govuk-!-font-weight-regular" style="padding: 5px 0px 6px 0px">
+                                Calculation reason
+                            </dt>
+                            <dd data-qa="calculation-reason" class="govuk-summary-list__value " style="padding: 5px 0px 6px 0px">
+                                {{ calculationReason }}
+                            </dd>
+                        </div>
+                    </dl>
+                {% endif %}
+
+                <h2 class="govuk-heading-m govuk-!-margin-bottom-2" data-qa="release-date-title">Release dates</h2>
+
+                {% if keyDatesArray and keyDatesArray.length > 0 %}
+                  {{ keyDates(keyDatesArray) }}
+                {% else %}
+                  <p>No key dates available.</p>
```

Comment:
possibly untested

Reply:
Added test case - GET /calculation/:nomsId/summary/:calculationRequestId/printNotificationSlip?fromPage=calculation should generate correct content for missing Agency Name, keyDates and Adjustments